test/math.js

```diff
@@ -1,10 +1,11 @@
 /*global describe, it, expect, require */
 
 var Assertion = expect().constructor;
-Assertion.prototype.almostEqual = function (obj, precision) {
+var MATH_EXP_ERROR = Math.abs(1 - Math.exp(704.6589) / 1.0702171200481775e+306) / Number.EPSILON;
+Assertion.prototype.almostEqual = function (expected) {
   'use strict';
-  var allowedDiff = precision || 1e-11;
-  return this.within(obj - allowedDiff, obj + allowedDiff);
+  var ok = Math.abs(1 - this.__flags.object / expected) / Number.EPSILON < MATH_EXP_ERROR + 8;
+  this.assert(ok, 'expected #{this} to be almost equal ' + expected, 'expected #{this} to not be almost equal' + expected);
 };
```

Comment:
this is not ideal, I do not know how to access private "this.__flags.object" and I did not test it well.